rosbag2_cpp/test/rosbag2_cpp/test_sequential_writer.cpp

```diff
@@ -386,3 +386,62 @@ TEST_F(SequentialWriterTest, snapshot_mode_zero_cache_size_throws_exception)
   std::string rmw_format = "rmw_format";
   EXPECT_THROW(writer_->open(storage_options_, {rmw_format, rmw_format}), std::runtime_error);
 }
+
+TEST_F(SequentialWriterTest, split_event_calls_callback)
+{
+  const int message_count = 7;
+  const int max_bagfile_size = 5;
+
+  ON_CALL(
+    *storage_,
+    write(An<std::shared_ptr<const rosbag2_storage::SerializedBagMessage>>())).WillByDefault(
+    [this](std::shared_ptr<const rosbag2_storage::SerializedBagMessage>) {
+      fake_storage_size_ += 1;
+    });
+
+  ON_CALL(*storage_, get_bagfile_size).WillByDefault(
+    [this]() {
+      return fake_storage_size_;
+    });
+
+  ON_CALL(*metadata_io_, write_metadata).WillByDefault(
+    [this](const std::string &, const rosbag2_storage::BagMetadata & metadata) {
+      fake_metadata_ = metadata;
+    });
+
+  ON_CALL(*storage_, get_relative_file_path).WillByDefault(
+    [this]() {
+      return fake_storage_uri_;
+    });
+
+  auto sequential_writer = std::make_unique<rosbag2_cpp::writers::SequentialWriter>(
+    std::move(storage_factory_), converter_factory_, std::move(metadata_io_));
+  writer_ = std::make_unique<rosbag2_cpp::Writer>(std::move(sequential_writer));
+
+  auto message = std::make_shared<rosbag2_storage::SerializedBagMessage>();
+  message->topic_name = "test_topic";
+
+  storage_options_.max_bagfile_size = max_bagfile_size;
+
+  bool callback_called = false;
+  std::string closed_file, opened_file;
+  rosbag2_cpp::bag_events::WriterEventCallbacks callbacks;
+  callbacks.write_split_callback =
+    [&callback_called, &closed_file, &opened_file](rosbag2_cpp::bag_events::BagSplitInfo & info) {
+      closed_file = info.closed_file;
+      opened_file = info.opened_file;
+      callback_called = true;
+    };
+  writer_->add_event_callbacks(callbacks);
+
+  writer_->open(storage_options_, {"rmw_format", "rmw_format"});
+  writer_->create_topic({"test_topic", "test_msgs/BasicTypes", "", ""});
+
+  for (auto i = 0; i < message_count; ++i) {
+    writer_->write(message);
+  }
+
+  ASSERT_TRUE(callback_called);
+  EXPECT_EQ(closed_file, storage_options_.uri + '/' + storage_options_.uri + "_0");
```

Comment:
This is why the Windows test is failing - to create URIs always use a filesystem library rather than assuming path separators in your code. For this you can use `rcpputils::fs::path` object, which supports the `/` operator for concatenation. Something like

```suggestion
  std::string expected_closed = (rcpputils::fs::path(storage_options_.uri) / (storage_options_.uri + "_0")).string();
  EXPECT_EQ(closed_file, expected_closed);
```